core/state/state_object.go

```diff
@@ -80,6 +80,9 @@ type stateObject struct {
 	dirtyCode bool // true if the code was updated
 	suicided  bool
 	deleted   bool
+
+	created   bool // true if the object was created in the current transaction
+	sendalled bool // true if the object executed SENDALL in the current tx
```

Comment:
I'm wondering if both of these are needed. 
Seems to me that `created` is needed, but other than that, we could use the `suicided` flag?
Because basically, the new rule is that we are allowed to set `suicided` `iff` `created=true`

Reply:
In that case, the statedb SendAll function could be basically
```
func (s *StateDB) SendAll(addr common.Address) bool {
	stateObject := s.getStateObject(addr)
	if stateObject == nil {
		return false
	}
	if stateObject.created{
		return s.Suicide(addr)
	}
```

?